src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -424,33 +428,40 @@ private static Expression CreateRouteHandlerInvocationContextBase(FactoryContext
         return fallbackConstruction;
     }
 
-    private static void AddTypeProvidedMetadata(MethodInfo methodInfo, List<object> metadata, IServiceProvider? services)
+    private static void AddTypeProvidedMetadata(MethodInfo methodInfo, List<object> metadata, IServiceProvider? services, Span<ParameterInfo> propertiesAsParameter)
     {
         object?[]? invokeArgs = null;
 
-        // Get metadata from parameter types
-        var parameters = methodInfo.GetParameters();
-        foreach (var parameter in parameters)
+        void AddMetadata(ReadOnlySpan<ParameterInfo> parameters)
```

Comment:
What do you think about instead having a local func to create a combined IEnumerable so you can loop over both methodInfo.GetParameters() and propertiesAsParameter at the same time?
```csharp
static IEnumerable<int> f(IEnumerable<int> i2, IEnumerable<int> j2)
{
    foreach (var elem in i2)
    {
        yield return elem;
    }
    foreach (var elem in j2)
    {
        yield return elem;
    }
}

foreach (var i in f(i1, i2))
{
    // Do metadata stuff
}
```

Reply:
That might be a good idea, let me check how better or worse it is.

Reply:
@BrennanConroy I benchmarked both approaches:

```C#
    [Benchmark]
    public void Merge()
    {
        static IEnumerable<string> MergeLists(IEnumerable<string> parameterInfos, IEnumerable<string> additionalParameterInfos)
        {
            foreach (var elem in parameterInfos)
            {
                yield return elem;
            }
            foreach (var elem in additionalParameterInfos)
            {
                yield return elem;
            }
        }

        foreach (var item in MergeLists(names, additionalNames))
        {}
    }
```

and 

``` c#
    [Benchmark]
    public void MultipleCalls()
    {
        void Work(ReadOnlySpan<string> data)
        {
            foreach (var item in data)
            {
            }
        }

        Work(names);
        Work(CollectionsMarshal.AsSpan(additionalNames));
    }
```

I know it is uglier, but the results shown that the current approach is much better.

|        Method |       Mean |     Error |    StdDev |  Gen 0 | Allocated |
|-------------- |-----------:|----------:|----------:|-------:|----------:|
|         Merge | 118.764 ns | 2.2431 ns | 2.3035 ns | 0.0114 |     144 B |
| MultipleCalls |   6.083 ns | 0.0493 ns | 0.0437 ns |      - |         - |

Reply:
Ah, I guess if you wanted to do it better you'd need to define a struct enumerable and use that, but it's a lot more code. Could we at least mark the local function as static and pass in any needed state?

Reply:
done.